src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue; // skip header
+            }
+            String[] parts = line.split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid line format: " + line);
+            }
+            Operation operation;
+            try {
+                operation = findOperation(parts[0]);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Unknown operation code: " + parts[0]);
+            }
+            String fruit = parts[1];
+            int quantity;
+            try {
+                quantity = Integer.parseInt(parts[2]);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Invalid quantity format: " + parts[2]);
+            }
+            FruitTransaction transaction = new FruitTransaction();
```

Comment:
Consider using a constructor or a static factory method in `FruitTransaction` to create instances instead of setting each field separately. This will make the code more concise and reduce the risk of missing out on setting any field.